smart_campus/app/views.py

```diff
@@ -84,17 +84,17 @@ def signup(request):
     """
     user_email = request.POST.get('email')
     password = request.POST.get('password')
-    nickname = request.POST.get('nickname', '')
+    nickname = request.POST.get('nickname')
 
-    if not user_email or not password:
-        return HttpResponse('Either email or password input is missing.', status=400)
+    if not user_email or not password or not nickname:
+        return HttpResponse('Either email, password or nickname input is missing.', status=404)
 
     try:
         user = User.objects.create_user(user_email, password, nickname)
     except (ValueError, ValidationError) as error:
         return HttpResponse(error, status=400)
     except IntegrityError:
-        return HttpResponse('The email is already taken, try another!', status=400)
+        return HttpResponse('The email is already taken, try another!', status=403)
```

Comment:
And use `409` in this case.

Reference
https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists